demoproductform/src/Form/Modifier/ProductFormModifier.php

```diff
@@ -0,0 +1,92 @@
+<?php
+/**
+ * 2007-2020 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Academic Free License 3.0 (AFL-3.0).
+ * It is also available through the world-wide-web at this URL: https://opensource.org/licenses/AFL-3.0
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\Module\DemoProductForm\Form\Modifier;
+
+use PrestaShop\Module\DemoProductForm\CQRS\CommandHandler\SaveMyModuleCustomFieldHandler;
+use PrestaShop\Module\DemoProductForm\Entity\CustomProduct;
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\ProductId;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Translation\TranslatorInterface;
+
+final class ProductFormModifier
+{
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @param TranslatorInterface $translator
+     */
+    public function __construct(
+        TranslatorInterface $translator
+    ) {
+        $this->translator = $translator;
+    }
+
+    /**
+     * @param ProductId|null $productId
+     * @param FormBuilderInterface $productFormBuilder
+     */
+    public function modify(?ProductId $productId, FormBuilderInterface $productFormBuilder): void
+    {
+        $basicTabFormBuilder = $productFormBuilder->get('basic');
+
+        $this->modifyBasicTab($productId, $basicTabFormBuilder);
+    }
+
+    /**
+     * @param ProductId|null $productId
+     * @param FormBuilderInterface $basicTabFormBuilder
+     *
+     * @see SaveMyModuleCustomFieldHandler to check how the field is handled on form POST
+     */
+    private function modifyBasicTab(?ProductId $productId, FormBuilderInterface $basicTabFormBuilder): void
+    {
+        $idValue = $productId ? $productId->getValue() : null;
+        $customProduct = new CustomProduct($idValue);
+
+        $formTypes = [];
+        /** @var FormBuilderInterface $formType */
+        foreach ($basicTabFormBuilder->all() as $formType) {
+            $typeName = $formType->getName();
+            // collect all the form child into local variable and remove them from
+            $formTypes[$typeName] = $formType;
+            $basicTabFormBuilder->remove($typeName);
+        }
+
+        foreach ($formTypes as $name => $type) {
```

Comment:
I wonder if we could have any helper class to do this for us 🤔

Reply:
https://github.com/PrestaShop/PrestaShop/pull/24337
Tried something here, but got stuck on how to smoothly integrate that modified builder